plugins/mtd/fu-mtd-device.c

```diff
@@ -35,6 +36,22 @@ fu_mtd_device_to_string(FuDevice *device, guint idt, GString *str)
 	fwupd_codec_string_append_hex(str, idt, "EraseSize", self->erasesize);
 	fwupd_codec_string_append_hex(str, idt, "MetadataOffset", self->metadata_offset);
 	fwupd_codec_string_append_hex(str, idt, "MetadataSize", self->metadata_size);
+	fwupd_codec_string_append_hex(str, idt, "IsPciDevice", self->is_pci_device);
+}
+
+static gchar *
+fu_mtd_device_convert_version(FuDevice *self, guint64 version_raw)
+{
+	/* let's assume for now that any PCI device with pair version format uses B&R encoding */
+	if (FU_MTD_DEVICE(self)->is_pci_device &&
```

Comment:
this is fine, but looks *weird*. Can you use `FuDevice *device` and then do `FuMtdDevice *self = FU_MTD_DEVICE(device);` pls.